cpp/src/arrow/util/future.h

```diff
@@ -154,9 +154,8 @@ class FutureStorage : public FutureStorageBase {
 
   Status status() const { return result_.status(); }
 
-  template <typename U>
-  void MarkFinished(U&& value) {
-    result_ = std::forward<U>(value);
+  void MarkFinished(Result<T> result) {
```

Comment:
Does this implicitly convert from any T-convertible U?

Reply:
Yes, because `Result<T>` already has that implicit constructor